### ISSUE TITLE
Quarkus 3 / Hibernate: HV000254: Missing parameter metadata for %s, which declares implicit or synthetic parameters

### ISSUE THREAD

**Issue by cthiebault** [quarkusio/quarkus#33726]
### Describe the bug

Hello,
I'm upgrading to Quarkus 3.0.4, and I'm getting the following error with Hibernate entities that use inheritance:

```txt
HV000254: Missing parameter metadata for %s, which declares implicit or synthetic parameters. 
Automatic resolution of generic type information for method parameters may yield incorrect 
results if multiple parameters have the same erasure. 
To solve this, compile your code with the -parameters flag.
```

I'm using Kotlin (1.8.21) with Gradle (8.1.1).
I've configured Gradle to compile Kotlin with the `-parameters` flag, but it doesn't seem to work:

**-- build.gradle.kts --**
```kotlin
tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
  kotlinOptions {
    jvmTarget = "17"
    javaParameters = true
  }
}
```

Here are my JPA entities:

```kotlin
@Entity
@Table(name = "shape")
@Inheritance(strategy = JOINED)
@Cacheable
abstract class ShapeJpa(

  @Id
  @JdbcTypeCode(SqlTypes.VARCHAR)
  @Column(name = "id", updatable = false, nullable = false, unique = true)
  open val id: UUID,

  @Length(max = 100)
  @Column(name = "name", nullable = false, length = 100)
  open var name: String,
)

@Entity
@Table(name = "rectangle_shape")
@PrimaryKeyJoinColumn(name = "shape_id")
data class RectangleJpa(
  override val id: UUID,
  override var name: String,
) : ShapeJpa(id, name) {

  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (javaClass != other?.javaClass) return false
    return id == (other as RectangleJpa).id
  }

  override fun hashCode(): Int = id.hashCode()
}
```

Any idea of what could be the problem?

I'll try to set up a minimal reproducible example, but with a very basic Quarkus project, I was not able to reproduce the
issue :-/

### Output of `uname -a` or `ver`

Linux cthiebault 6.1.30-1-MANJARO #1 SMP PREEMPT_DYNAMIC Wed May 24 22:51:44 UTC 2023 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.7" 2023-04-18 OpenJDK Runtime Environment Temurin-17.0.7+7 (build 17.0.7+7) OpenJDK 64-Bit Server VM Temurin-17.0.7+7 (build 17.0.7+7, mixed mode, sharing)

### Quarkus version or git rev

3.0.4

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.1.1

### Additional information

------------------------------------------------------------
Gradle 8.1.1
------------------------------------------------------------

Build time:   2023-04-21 12:31:26 UTC
Revision:     1cf537a851c635c364a4214885f8b9798051175b

Kotlin:       1.8.10
Groovy:       3.0.15
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          17.0.7 (Eclipse Adoptium 17.0.7+7)
OS:           Linux 6.1.30-1-MANJARO amd64

**Comment by cthiebault** [comment]
Thanks for your response.

I've created a repo that reproduce the bug of Hibernate entities with inheritance with Kotlin:
https://github.com/cthiebault/quarkus-hibernate-kotlin

It works well with Quarkus  2 (`quarkus2` branch) but not with Quarkus 3 (`quarkus3` branch).

I don't see this `HV000254` warning anymore but inheritance does not work...

**Comment by yrodiere** [comment]
> I don't see this HV000254 warning anymore but inheritance does not work...

From what I understand this is does not reproduce the issue you reported, then... ? It's just another issue?

**Comment by cthiebault** [comment]
Good point :-)
In fact I had issues using Hibernate inheritance with Kotlin and as I saw this `HV000254` warning I thought it was because of this...
My bad: I created an issue about a warning message and not about my real problem :-(
Do you want me to create a new issue?

**Comment by Sanne** [comment]
> I don't see this HV000254 warning anymore but inheritance does not work...

Right that Logger line existed in Validator 8.0.0.Final but got removed already, so there's nothing to fix, besides possibly upgrading Validator in Quarkus.

@cthiebault if it's a different issue, please open a new issue.

**Comment by Sanne** [comment]
> About the warning, can either @Sanne or @gsmet confirm it's innocuous and will disappear once we upgrade HV in Quarkus? If so let's close this issue.

It will certainly disappear once we upgrade HV in Quarkus; but I don't know if a release is being planned, that's up to @gsmet to decide.

**Comment by cthiebault** [comment]
I've created https://github.com/quarkusio/quarkus/issues/33740 for the Hibernate inheritance & Kotlin issue.
Thanks for help!

**Comment by gsmet** [comment]
I'm not sure the setup is correct as I would expect the message to be properly logged (and not contain `%s`).

Are you using our tooling to generate the Gradle project? Because I would expect things to work properly when generating a project with our tooling.

**Comment by cthiebault** [comment]
> Are you using our tooling to generate the Gradle project? Because I would expect things to work properly when generating a project with our tooling.

Yes I've compared my Gradle config in (https://github.com/cthiebault/quarkus-hibernate-kotlin) with the code generated by https://code.quarkus.io/ and it's well configured.

**Comment by geoand** [comment]
What should be done with this issue?

**Comment by yrodiere** [comment]
@geoand Well you're asking about Hibernate Validator on the very first day of Guillaume's PTO, so... I guess you'd better try again later :)

**Comment by gsmet** [comment]
We fixed that in HV 8.0.1 so we should be all good.

**Comment by geoand** [comment]
Thanks for the update!
